### PR TITLE
DX-1264: Update Exports & Add CI Tests

### PR DESCRIPTION
**Pull request by ytkimirti**
This PR has three main changes:
- update eslint & exports in package.json
- fixing small errors in examples and adding extensive CI
- llamaindex is removed, as it was causing errors explained below

Since llamaindex is removed, we will release a major version (2.0.0)

## Eslint & Exports

- Updated tsup config
- Updated exports field in package.json
>NOTE: This should not bring any breaking changes with the old version. Here is the old and new output coming from the attw cli:
rag before
rag after

## CI

The examples are first tested with a local build. Then, a release is made with ci tag and it's released. It's tested by deploying to Vercel and CF workers.

### CODE REVIEW

File: examples/cloudflare-workers/README.md [hunk 1]
```diff
@@ -28,13 +28,7 @@ npm install @upstash/rag-chat
 Ensure your wrangler.toml file includes the following configuration to enable Node.js compatibility:
 
 ```toml
-node_compat = true
-```
-
-In older CF worker versions, you may need to set the following compatibility flags:
-
-```toml
-compatibility_flags = [ "streams_enable_constructors", "transformstream_enable_standard_constructor" ]
```
Comment (CahidArda) [inline examples/cloudflare-workers/README.md:37]: why do we delete these? I needed them for streaming, did you try streaming with only the nodejs_compat_v2 flag?

Comment (ytkimirti) [reply examples/cloudflare-workers/README.md:37]: Yes it works, why did you need them, shouldn't they be default?
https://developers.cloudflare.com/workers/configuration/compatibility-dates/#streams-constructors
https://developers.cloudflare.com/workers/configuration/compatibility-dates/#compliant-transformstream-constructor

Comment (CahidArda) [reply examples/cloudflare-workers/README.md:37]: `In older CF worker versions` :D

let's delete them then, if someone has any issues, we can let them know about the flags.

File: examples/cloudflare-workers/src/index.ts [hunk 2]
```diff
@@ -1,8 +1,117 @@
...
+    UPSTASH_VECTOR_REST_URL: string;
+    UPSTASH_VECTOR_REST_TOKEN: string;
+    QSTASH_TOKEN: string;
+    OPENAI_API_KEY: string;
+  };
+}>();
+
+app.use("*", async (c, next) => {
+  const ragChat = new RAGChat({
+    // model: upstash("meta-llama/Meta-Llama-3-8B-Instruct", { apiKey: c.env.QSTASH_TOKEN }),
```
Comment (CahidArda) [inline examples/cloudflare-workers/src/index.ts:21]: we can put the model back in, and comment out the alternative below

File: examples/cloudflare-workers/README.md (hunk 1 above)
Comment (ytkimirti) [reply examples/cloudflare-workers/README.md:37]: Ohh makes sense, I will add it back

File: package.json [hunk 3]
```diff
@@ -71,19 +77,17 @@
     "llamaindex": "0.6.0",
     "nanoid": "^5.0.7",
     "pdf-parse": "^1.1.1",
-    "unstructured-client": "^0.15.1"
+    "unstructured-client": "^0.15.1",
+    "@langchain/anthropic": "^0.2.15",
+    "@langchain/mistralai": "^0.0.28",
+    "langsmith": "^0.1.41"
```
[the remark this answers is not in the document]
Comment (ytkimirti) [reply package.json:83]: tsup does not bundle dependencies by default so we don't have to label them as external. these will just be normal dependencies.

We can turn them into optional peer dependencies in the future, that's probably the best way to do it but we need dynamic imports for that. And ofc that would be a breaking change so in the future

File: tsup.config.ts [hunk 4]
```diff
@@ -4,32 +4,21 @@ export default defineConfig([
...
     format: ["cjs", "esm"],
-    sourcemap: false,
     clean: true,
     dts: true,
-    minify: false,
-    external: ["@langchain/anthropic", "@langchain/mistralai", "langsmith"],
   },
   {
-    entry: ["src/nextjs/rsc-server.ts", "src/nextjs/rsc-client.ts"],
+    entry: ["src/nextjs/rsc-server.ts"],
```
[the remark this answers is not in the document]
Comment (ytkimirti) [reply tsup.config.ts:11]: yepp, actually there is no special key for rsc-client. Only special import is for "react-server" so we actually can't make react use "rsc-client.ts" and normal imports to use "index.ts"

File: src/nextjs/index.ts [hunk 5]
```diff
@@ -1,3 +1,7 @@
-export type { createServerActionStream } from "./rsc-server";
-export type { readServerActionStream } from "./rsc-client";
-export type { aiUseChatAdapter } from "./chat-adapter";
+export * from "./chat-adapter";
+export * from "./server-action-read-adapter";
+
+// This cant be exported here because you can't import ai/rsc
+// when not using a server component
+// Opposite is true for the read adapter, it can't be imported in the rsc-server.ts file
```
[the remark this answers is not in the document]
Comment (ytkimirti) [reply src/nextjs/index.ts:6]: Yeah, I should make a better explanation in the commet

File: examples/cloudflare-workers/src/index.ts (hunk 2 above)
Comment (ytkimirti) [reply examples/cloudflare-workers/src/index.ts:21]: Oh i missed this thanks

**Comment by CahidArda** [comment]
[more on react-server export](https://github.com/reactjs/rfcs/blob/main/text/0227-server-module-conventions.md)

**Comment by CahidArda** [comment]
✘ [ERROR] The `nodejs_compat_v2` compatibility flag is experimental and must be prefixed with `experimental:`. Use `experimental:nodejs_compat_v2` flag instead.

**Comment by CahidArda** [comment]
managed to reproduce the `Error: Missing tiktoken_bg.wasm` error (mentioned in https://github.com/upstash/rag-chat/issues/82) in the CI test of [20e3c11](https://github.com/upstash/rag-chat/pull/79/commits/20e3c11f4f1a282cf0e005eacd96387de78ae444).

[See](https://github.com/upstash/rag-chat/actions/runs/11147944037/job/30991074857)

**Comment by CahidArda** [comment]
Some notes on the error:
- the error we get depends on the package manager used in the app (npm vs pnpm)

| - | - | - | - |
| rag-chat version | llamaindex version | Vercel (npm) | Vercel (pnpm) |
| 1.6.0 | ^0.5.20 | ❌ llamaindex/openai version missing | I didn't try. Probably the same error as the one to the left |
| 1.6.1 | ^0.6.0 | ✅ | ❌ unexpected char |
| 1.6.4 | 0.5.24 | ✅ | ❌ [Missing tiktoken_bg.wasm](https://github.com/run-llama/create-llama/issues/164) |

the missing file has a task at Vercel https://github.com/vercel/next.js/issues/54395 and at tiktoken https://github.com/dqbd/tiktoken/issues/67.

The unexpected char problem:
- referenced in @xenova/transformers https://github.com/xenova/transformers.js/issues/210 and they suggest updating dependencies. I tried both dependencies and peerDependencies. It didn't work.
- potential solutions in https://github.com/xenova/transformers.js/issues/338. serverComponentsExternalPackages ([1](https://github.com/xenova/transformers.js/issues/210#issuecomment-1886070014), [2](https://github.com/xenova/transformers.js/issues/338#issuecomment-1742713301)) and [pages router](https://github.com/xenova/transformers.js/issues/338#issuecomment-1743921037).

["pnpm is a bad replacement"](https://github.com/xenova/transformers.js/issues/875#issuecomment-2260597579)

**Comment by CahidArda** [comment]
let's squash merge

**Comment by mrsamirr** [comment]
well @CahidArda,
| rag-chat version | llamaindex version | Vercel (npm) |
|------------------|--------------------|--------------|
| 1.6.4            | 0.5.24             | ❌ [missing tiktoken_bg.wasm](https://github.com/run-llama/create-llama/issues/164) |

The issue with `missing tiktoken_bg.wasm` on Vercel has been resolved as mentioned in [this comment](https://github.com/run-llama/create-llama/issues/164#issuecomment-2387573668). After resolving this issue, the rag-chat application now works seamlessly on Vercel.

**Comment by CahidArda** [comment]
Hey @mrsamirr,

we can reconsider adding llama parser to rag-chat in the future, but we didn't want to make installing rag-chat require that much configuration at this time.

Thanks a lot on reporting your solution!

**Pull request merged**